packages/image_picker/image_picker_macos/lib/image_picker_macos.dart

```diff
@@ -141,30 +202,77 @@ class ImagePickerMacOS extends CameraDelegatingImagePickerPlatform {
   }
 
-  // `maxWidth`, `maxHeight`, and `imageQuality` arguments are not currently
-  // supported. If any of these arguments are supplied, they will be silently
-  // ignored.
+  // This is soft-deprecated in the platform interface, and is only implemented
+  // for compatibility. Callers should be using getMultiImageWithOptions.
   @override
   Future<List<XFile>> getMultiImage({
     double? maxWidth,
     double? maxHeight,
     int? imageQuality,
   }) async {
-    // TODO(stuartmorgan): Add a native implementation that can use
-    // PHPickerViewController on macOS 13+, with this as a fallback for
-    // older OS versions: https://github.com/flutter/flutter/issues/125829.
+    return getMultiImageWithOptions(
+      options: MultiImagePickerOptions(
+        imageOptions: ImageOptions(
+          imageQuality: imageQuality,
+          maxHeight: maxHeight,
+          maxWidth: maxWidth,
+        ),
+      ),
+    );
+  }
+
+  // [MultiImagePickerOptions] options are currently only
+  // supported when using PHPicker implementation. If any of these arguments are supplied, they will be silently
+  // ignored.
+  @override
+  Future<List<XFile>> getMultiImageWithOptions({
+    MultiImagePickerOptions options = const MultiImagePickerOptions(),
+  }) async {
+    if (await shouldUsePHPicker()) {
+      final List<String> images = await _hostApi.pickImages(
+        _imageOptionsToImageSelectionOptions(options.imageOptions),
+        GeneralOptions(
+          limit: options.limit ?? 0,
+        ),
+      );
+      return images.map((String imagePath) => XFile(imagePath)).toList();
+    }
     const XTypeGroup typeGroup =
         XTypeGroup(uniformTypeIdentifiers: <String>['public.image']);
     final List<XFile> files = await fileSelector
         .openFiles(acceptedTypeGroups: <XTypeGroup>[typeGroup]);
     return files;
   }
 
-  // `maxWidth`, `maxHeight`, and `imageQuality` arguments are not currently
-  // supported. If any of these arguments are supplied, they will be silently
+  ImageSelectionOptions _imageOptionsToImageSelectionOptions(
+    ImageOptions imageOptions,
+  ) {
+    return ImageSelectionOptions(
+      quality: imageOptions.imageQuality ?? 100,
+      maxSize: MaxSize(
+        width: imageOptions.maxWidth,
+        height: imageOptions.maxHeight,
+      ),
+    );
+  }
+
+  // [ImageOptions] options are currently only
+  // supported when using PHPicker implementation. If any of these arguments are supplied, they will be silently
   // ignored.
   @override
   Future<List<XFile>> getMedia({required MediaOptions options}) async {
+    if (await shouldUsePHPicker()) {
+      final List<String> images = await _hostApi.pickMedia(
+        MediaSelectionOptions(
+          imageSelectionOptions:
+              _imageOptionsToImageSelectionOptions(options.imageOptions),
+        ),
+        GeneralOptions(
+          limit: options.limit ?? (options.allowMultiple ? 0 : 1),
```

Comment:
This seems to be incorrect, and needs validation [similarly to image_picker_android](https://github.com/flutter/packages/blob/main/packages/image_picker/image_picker_android/lib/image_picker_android.dart#L275-L318) since the current code doesn't check `allowMultiple` in case `limit` was not provided.

Reply:
It looks like [there is validation from the app-facing package](https://github.com/flutter/packages/blob/d8e8e8cee712b4bea345b804c9fc88df229ce154/packages/image_picker/image_picker/lib/image_picker.dart#L246-L265) (`image_picker`) for all methods, do we still need to verify from the platform implementation side?